packages/server/src/integrations/snowflake.ts

```diff
@@ -0,0 +1,122 @@
+import { Integration, QueryTypes, SqlQuery } from "../definitions/datasource"
+import {
+  SnowflakeError,
+  Statement,
+  createConnection,
+  Connection,
+} from "snowflake-sdk"
+
+module SnowflakeModule {
+  interface SnowflakeConfig {
+    account: string
+    username: string
+    password: string
+    warehouse: string
+    database: string
+    schema: string
+  }
+
+  const SCHEMA: Integration = {
+    docs: "https://developers.snowflake.com/",
+    description:
+      "Snowflake is a solution for data warehousing, data lakes, data engineering, data science, data application development, and securely sharing and consuming shared data.",
+    friendlyName: "Snowflake",
+    datasource: {
+      account: {
+        type: "string",
+        required: true,
+      },
+      username: {
+        type: "string",
+        required: true,
+      },
+      password: {
+        type: "password",
+        required: true,
+      },
+      warehouse: {
+        type: "string",
+        required: true,
+      },
+      database: {
+        type: "string",
+        required: true,
+      },
+      schema: {
+        type: "string",
+        required: true,
+      },
+    },
+    query: {
+      create: {
+        type: QueryTypes.SQL,
+      },
+      read: {
+        type: QueryTypes.SQL,
+      },
+      update: {
+        type: QueryTypes.SQL,
+      },
+      delete: {
+        type: QueryTypes.SQL,
+      },
+    },
+  }
+
+  class SnowflakeIntegration {
+    private client: Connection
+
+    constructor(config: SnowflakeConfig) {
+      this.client = createConnection(config)
+    }
+
+    async connectAsync() {
+      return new Promise((resolve, reject) => {
+        this.client.connect(function (err: any, conn: any) {
+          if (err) reject(err)
```

Comment:
Small change, but with `reject`, `resolve` like this I prefer to have braces around everything as its a little less ambigious when reading, e.g.
```
if (err) {
  reject(err)
}
resolve()
```
that way theres no risk of say the `reject(err)` being removed and suddenly the error state now resolves the call.

Also, if you wanted to avoid the need to wrap the SDK with promises, you could use https://www.npmjs.com/package/snowflake-promise

Reply:
snowflake-promise is much nicer! Wish I'd seen it in the first place!